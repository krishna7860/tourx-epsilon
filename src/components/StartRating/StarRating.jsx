/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable react/prop-types */
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import React, { useEffect, useState } from "react";
import { Star, StarHalf, StarOutline } from "react-ionicons";
import { uuid } from "uuidv4";
import PropTypes from "prop-types";
import { Wrapper } from "./Style";

function StarRating({ rating, width, height, color }) {
  const [state, setState] = useState({
    full: 0,
    half: 0,
    empty: 5,
  });
  useEffect(() => {
    const abs = Math.trunc(rating);
    const decimal = 5 - rating;

    const full = abs;
    const half = decimal ? 1 : 0;
    const empty = full + half === 5 ? 0 : 5 - (full + half);
    setState({
      full,
      half,
      empty,
    });
  }, [rating]);
  return (
    <Wrapper>
      {state.full &&
        new Array(state.full).fill(0).map((_i) => {
          return (
            <Star color={color} width={width} height={height} key={uuid()} />
          );
        })}
      {state.half
        ? new Array(state.half).fill(0).map((i) => {
            return (
              <StarHalf
                color={color}
                width={width}
                height={height}
                key={uuid()}
              />
            );
          })
        : null}
      {state.empty
        ? new Array(state.empty).fill(0).map((i) => {
            return (
              <StarOutline
                color={color}
                width={width}
                height={height}
                key={uuid()}
              />
            );
          })
        : null}
    </Wrapper>
  );
}

StarRating.propTypes = {
  rating: PropTypes.number.isRequired,
};

export default StarRating;
