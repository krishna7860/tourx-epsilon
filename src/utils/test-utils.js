/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from "react";
import { render as rtlRender } from "@testing-library/react";
// import { renderHook as rtlRenderHooks } from "@testing-library/react-hooks";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import * as themes from "../theme/schema.json";
import reducer from "../redux/rootReducer";
// import defaultStore from "../redux/defaultStore";

function render(ui, options = {}) {
  const {
    initialState = {},
    store = createStore(reducer, initialState),
    theme,
    ...renderOptions
  } = options;

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <ThemeProvider theme={themes.data.light}>{children}</ThemeProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// function renderHookWithRedux(callback, options = {}) {
//   const { initialState = defaultStore, store = createStore(reducer, initialState), ...others } = options

//   const returned = rtlRenderHooks(callback, {
//     ...others,
//     // eslint-disable-next-line react/display-name
//     wrapper: ({ children }) => <Provider store={store}>{children}</Provider>
//   })

//   return {
//     ...returned,
//     store
//   }
// }

// re-export everything
export * from "@testing-library/react";
// override render method
export { render, rtlRender as renderHook };
