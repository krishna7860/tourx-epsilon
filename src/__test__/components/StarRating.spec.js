/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { render } from "../../utils/test-utils";
import StarRating from "../../components/StartRating/StarRating";

describe("components/StarRating", () => {
  it("should be rendered properly", () => {
    const { asFragment } = render(
      <StarRating rating={3.5} color="#C0980A" width="15px" height="15px" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
