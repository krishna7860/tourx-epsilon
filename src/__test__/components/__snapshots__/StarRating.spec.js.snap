// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/StarRating should be rendered properly 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT eUTdid"
  >
    <span
      class="sc-bdnxRM hQwsEM"
      color="#C0980A"
      height="15px"
      rotate="0"
      width="15px"
    >
      <svg
        class=""
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M394 480a16 16 0 01-9.39-3L256 383.76 127.39 477a16 16 0 01-24.55-18.08L153 310.35 23 221.2a16 16 0 019-29.2h160.38l48.4-148.95a16 16 0 0130.44 0l48.4 149H480a16 16 0 019.05 29.2L359 310.35l50.13 148.53A16 16 0 01394 480z"
        />
      </svg>
    </span>
    <span
      class="sc-bdnxRM hQwsEM"
      color="#C0980A"
      height="15px"
      rotate="0"
      width="15px"
    >
      <svg
        class=""
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M394 480a16 16 0 01-9.39-3L256 383.76 127.39 477a16 16 0 01-24.55-18.08L153 310.35 23 221.2a16 16 0 019-29.2h160.38l48.4-148.95a16 16 0 0130.44 0l48.4 149H480a16 16 0 019.05 29.2L359 310.35l50.13 148.53A16 16 0 01394 480z"
        />
      </svg>
    </span>
    <span
      class="sc-bdnxRM hQwsEM"
      color="#C0980A"
      height="15px"
      rotate="0"
      width="15px"
    >
      <svg
        class=""
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M394 480a16 16 0 01-9.39-3L256 383.76 127.39 477a16 16 0 01-24.55-18.08L153 310.35 23 221.2a16 16 0 019-29.2h160.38l48.4-148.95a16 16 0 0130.44 0l48.4 149H480a16 16 0 019.05 29.2L359 310.35l50.13 148.53A16 16 0 01394 480z"
        />
      </svg>
    </span>
    <span
      class="sc-bdnxRM hQwsEM"
      color="#C0980A"
      height="15px"
      rotate="0"
      width="15px"
    >
      <svg
        class=""
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
          fill="none"
          stroke="currentColor"
          stroke-linejoin="round"
          stroke-width="32"
        />
        <path
          d="M256 48v316L118 464l54-160-140-96h172l52-160z"
        />
      </svg>
    </span>
    <span
      class="sc-bdnxRM hQwsEM"
      color="#C0980A"
      height="15px"
      rotate="0"
      width="15px"
    >
      <svg
        class=""
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
          fill="none"
          stroke="currentColor"
          stroke-linejoin="round"
          stroke-width="32"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
